rfd/0089-merge-webapps.md

```diff
@@ -0,0 +1,185 @@
+---
+authors: Jeff Pihach (jeff.pihach@goteleport.com)
+state: draft
+---
+
+# RFD 89 - Merge Webapps into Teleport
+
+## What
+
+- Merge the [webapps](https://github.com/gravitational/webapps) repo into the
+  [teleport](https://github.com/gravitational/teleport) repo.
+- Merge the [webapps.e](https://github.com/gravitational/webapps.e) repo into
+  the [teleport.e](https://github.com/gravitational/teleport.e) repo.
+
+## Why
+
+The version of the web UI and the Teleport are tightly coupled but the
+repositories were created separate to support both Gravity and Teleport. This
+introduced complexities when integrating the two projects.
+
+We have to separately clone and build the web UI and commit the webasset
+changes into the teleport repo whenever we want to represent an update. We also
+have to maintain multiple build systems and ops procedures for the web UI.
+
+Now that Gravity is no longer a product we develop, moving the web UI
+repositories into their respective Teleport repo will save us time and
+confusion.
+
+## Success Criteria
+
+Both [Teleport](https://github.com/gravitational/teleport) and
+[Teleport.e](https://github.com/gravitational/teleport.e) repos contain their
+respective UI counterparts and the build systems correctly build all projects.
+
+## Developer Experience
+
+### Day to day
```

Comment:
# How this will affect Connect

Generally not much as Connect is pretty self-contained. This is what comes to my mind on top of what Zac said.

## Default path to tsh

We'll have to update [the default path to tsh in dev mode](https://github.com/gravitational/webapps/blob/27c615b3ff6f317a85fac4aa28b8e73fa4aa0d28/packages/teleterm/src/mainProcess/runtimeSettings.ts#L18-L23). It's a simple change.

## Proto files

As mentioned by Zac, once webapps is merged into teleport, we want to streamline proto generation for Connect (#15955). Currently we update the protos in teleport, commit them there and then copy them over to webapps and commit the changes there as well. I suspect what we'll want to do is tell `buf` to generate Go proto files `lib/teleterm/api/protogen/golang` and the JS proto files inside `webapps/packages/teleterm/src/services/tshd/v1/`.

I'm not sure if #15955 needs to be implemented within this PR. It includes a lot of work already and we can live with copying the proto files for now.

Reply:
Oh sorry, #15955 is about an even more obscure issue with protos in Connect. What I had in mind were the tshd protos generated with `make grpc-teleterm`, [as described in Connect's readme](https://github.com/gravitational/webapps/tree/27c615b3ff6f317a85fac4aa28b8e73fa4aa0d28/packages/teleterm#generating-tshd-grpc-protobuf-files).

We don't have to worry about #15955 at all, nor about Connect protos in general. We'll just keep copying them over, the location will just change slightly. Once the merge is completed we can tweak `make grpc-teleterm` so that copying is not necessary. I'll make an issue about it as we already have a bunch of issues related to Connect's proto files.

Reply:
We should also add make targets for Teleport Connect that respect our existing OS/ARCH vars and correctly express the dependency on `build/tsh`.